main.rb

```diff
@@ -0,0 +1,121 @@
+module Enumerable
+  # my_each()
+  def my_each()
+    each do |item|
+      yield item if block_given?
+    end
+  end
+  # ending my_each()
+
+  # my_each_with_index()
+  def my_each_with_index()
+    index = 0
+    each do |item|
+      yield(item, index) if block_given?
+      index += 1
+    end
+  end
+  # my_each_with_index()
+
+  # my_select
+  def my_select()
+    new_arr = []
+    each do |item|
+      new_arr << item if yield(item)
+    end
+    new_arr
+  end
+  # my_select
+
+  # my_all?
+  def my_all?()
+    flag = true
+    each do |item|
+      feedback = yield(item)
+      flag = false if (feedback == false) || feedback.nil?
+    end
+    flag
+  end
+  # my_all?
+
+  # my_any
+  def my_any?()
+    flag = false
+    each do |item|
+      flag = true if yield(item)
+    end
+    flag
+  end
+  # my_any
+
+  # my_none?
+  def my_none?()
+    is_none = true
+    each do |item|
+      is_none = false if yield(item)
+    end
+    is_none
+  end
+  # my_none?
+
+  # my_count
+  def my_count
+    index = 0
+    each do |_item|
+      index += 1
+    end
+    index
+  end
+  # my_count
+
+  # my_map
+  def my_map(proc = nil, &block)
+    new_arr = []
+
+    if proc.is_a? Proc
+      each do |item|
+        new_item = proc.call(item)
+        new_arr << new_item
+      end
+      return new_arr
+    end
+
+    each do |item|
+      new_item = block.call(item)
+      new_arr << new_item
+    end
+    new_arr
+  end
+  # my_map
+
+  # my_inject
+  def my_inject(sign)
+    result = 0 if (sign == :+) || (sign == :-)
+    result = 1 if (sign == :*) || (sign == :/)
+    each do |item|
+      result = result.send(sign, item)
+    end
+    result
+  end
```

Comment:
- [ ] my_inject raises a "LocalJumpError" when no block or argument is given

- [ ] my_inject searches for the longest word in an array of strings

- [ ] my_inject does not mutate the original array

- [ ] my_inject when a block is given without an initail value combines all elements of enum by applying a binary operation, specified by a block

- [ ] my_inject when a block is given without an initail value combines all elements of enum by applying a binary operation, specified by a block::range

- [ ] my_inject when a block is given with an initail value combines all elements of enum by applying a binary operation, specified by a block

- [ ] my_inject when a symbol is specified with an initail value combines each element of the collection by applying the symbol as a named method

- [ ] my_inject when a symbol is specified with an initail value combines each element of the collection by applying the symbol as a named method::range